gidgethub/routing.py

```diff
@@ -79,6 +78,12 @@ async def dispatch(self, event: sansio.Event, *args: Any, **kwargs: Any) -> None
                 if data_key in event.data:
                     event_value = event.data[data_key]
                     if event_value in data_values:
-                        found_callbacks.extend(data_values[event_value])
+                        found_callbacks.update(data_values[event_value])
+        return found_callbacks
+
+    async def dispatch(self, event: sansio.Event, *args: Any, **kwargs: Any) -> None:
+        """Dispatch an event to all registered function(s)."""
+
+        found_callbacks = self.fetch(event)
```

Comment:
This requires bumping the major version number as this changes the execution order of routes.

Reply:
#136 is suggesting a minor version update.

Reply:
If this PR needs a major version bump, perhaps I can go ahead and release whatever we have on `master` as 4.2.0, so this PR can become 5.0.0.

Reply:
@dhruvmanila that true _if_ nothing changed warranting a version bump by this PR. But you are changing the semantics of how routers work, so that will require a major version bump (we are not using CalVer or something similar where semantic changes are not reflected in the version number).

@Mariatta 👍

Reply:
@brettcannon Done ✅